src/Microsoft.Health.Fhir.Azure/ExportDestinationClient/AzureExportDestinationClient.cs

```diff
@@ -93,7 +94,9 @@ public async Task CommitAsync(CancellationToken cancellationToken)
         {
             CheckIfClientIsConnected();
 
-            var uploadAndCommitTasks = new List<Task>();
+            // Upload all blocks for each blob that was modified.
+            var uploadTasks = new List<Task>();
```

Comment:
nit: specify the initial capacity so the list doesn't have to grow. Also, use the array instead of list here since you know the size and it's cheaper to allocate.